lib/vdsm/gluster/cli.py

```diff
@@ -1149,17 +1149,29 @@ def _parseGeoRepStatus(tree):
         for session in volume.findall('sessions/session'):
             pairs = []
             sessionDetail = {}
-            sessionDetail['sessionKey'] = session.find('session_slave').text
+            if session.find('session_secondary') is None:
```

Comment:
Maybe it would be a good idea to look up the format only once, outside the nested loops, and store it into a variable used here and below to select the proper format.

Reply:
I tought so at the begining, but i'm still learning Python language. In fact, this is my first real projet using Python and I dont know API and functions very well yet. However I learnt about XML parsing and updated the code to identify the version before entering the loop.

Reply:
Cool :-), one more thing to fix and then the patch should be ready.